internal/cmd/gocdk/_static/demo/blob.bucket/demo_blob_bucket.go

```diff
@@ -71,101 +90,193 @@ const blobBucketBaseTemplate = `
     <li><a href="./list">List</a> the contents of the bucket</li>
     <li><a href="./view">View</a> the contents of a specific blob in the bucket</li>
     <li><a href="./write">Write</a> a new blob into the bucket</li>
-  </ul>
+</ul>
+{{if .Err}}
+  <p><strong>{{.Err}}</strong></p>
+{{end}}`
+
+	blobBucketTemplateSuffix = `
 </body>
-</html>
-`
+</html>`
 
-func blobBucketBaseHandler(w http.ResponseWriter, req *http.Request) {
-	tmpl := template.Must(template.New("base").Parse(blobBucketBaseTemplate))
-	if err := tmpl.Execute(w, bucketURL); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-	}
-}
+	// Input: *blobBucketBaseData.
+	blobBucketBaseTemplate = blobBucketTemplatePrefix + blobBucketTemplateSuffix
 
-// Input: []*blob.ListObject.
-const blobBucketListTemplate = `
-<!DOCTYPE html>
-<html>
-<head>
-  <meta charset="UTF-8">
-  <title>blob.Bucket demo</title>
-</head>
-<body>
-  {{range .}}
+	// Input: *blobBucketListData.
+	blobBucketListTemplate = blobBucketTemplatePrefix + `
+  {{range .ListObjects}}
     <div>
       {{if .IsDir}}
         <a href="./list?prefix={{ .Key }}">{{ .Key }}</a>
       {{else}}
         <a href="./view?key={{ .Key }}">{{ .Key }}</a>
       {{end}}
     </div>
+  {{end}}` + blobBucketTemplateSuffix
+
+	// Input: *blobBucketViewData.
+	blobBucketViewTemplate = blobBucketTemplatePrefix + `
+  {{if .ListObjects}}
+    <form>
+    Choose a blob to view:
+    <select name="key">
+      {{range .ListObjects}}
+        <option value="{{.Key}}">{{.Key}}</option>
+      {{end}}
+    </select>
+    <br/>
+    <input type="submit">
+    </form>
+  {{end}}` + blobBucketTemplateSuffix
+
+	// Input: *blobBucketWriteData.
+	blobBucketWriteTemplate = blobBucketTemplatePrefix + `
+  {{if .WriteSuccess}}
+    Wrote it!
   {{else}}
-    <div>no blobs in bucket</div>
-  {{end}}
-</body>
-</html>
-`
+    <form>
+    Blob key to write to (any previous blob will be overwritten):
```

Comment:
The explanations for the input fields should ideally be labels, but at least grouped by semantic html for accessibility.

Reply:
Added labels.